src/zenml/cli/stack_components.py

```diff
@@ -409,6 +421,71 @@ def update_stack_component_command(
     return update_stack_component_command
 
 
+def generate_stack_component_remove_attribute_command(
+    component_type: StackComponentType,
+) -> Callable[[str, List[str]], None]:
+    """Generates an `remove_attribute` command for the specific stack component type."""
+    display_name = _component_display_name(component_type)
+
+    @click.argument(
+        "name",
+        type=str,
+        required=True,
+    )
+    @click.argument("args", nargs=-1, type=click.UNPROCESSED)
+    def remove_attribute_stack_component_command(
+        name: str, args: List[str]
+    ) -> None:
+        """Removes one or more attributes from a stack component."""
+        cli_utils.print_active_profile()
+        with console.status(f"Updating {display_name} '{name}'...\n"):
+            repo = Repository()
+            current_component = repo.get_stack_component(component_type, name)
+            if current_component is None:
+                cli_utils.error(f"No {display_name} found for name '{name}'.")
+
+            try:
+                parsed_args = cli_utils.parse_unknown_component_attributes(args)
+            except AssertionError as e:
+                cli_utils.error(str(e))
+                return
+
+            component_class = repo.get_flavor(
+                name=current_component.FLAVOR,
+                component_type=component_type,
+            )
+            required_attributes = current_component.schema()["required"]
+            for arg in parsed_args:
+                if (
+                    arg in required_attributes
+                    or arg in MANDATORY_COMPONENT_ATTRIBUTES
+                ):
+                    cli_utils.error(
+                        f"Cannot remove mandatory attribute '{arg}' of '{name}' {current_component.TYPE}. "
+                    )
```

Comment:
Indeed I wondered about this very point. The reason why I split up the loops was so that the error messages could be different. i.e. one is preventing the removal of a mandatory attribute, and the other is only allowing the removal of optional attributes. I feel like it's more helpful for users to have different error messages rather than just a generic 'you did something wrong'. WDYT?

Reply:
Personally I think saying that the attribute they're trying to remove is not one of the optional ones (and printing a list of optional ones) would be clear enough, but your solution is even more detailed and I'm not opposed to having it in.

Let me make a suggestion still that at least avoids the double for loop:
```python
optional_attributes = _get_optional_attributes(current_component.__class__)
required_attributes = _get_required_attributes(current_component.__class__)

for arg in parsed_args:
    if arg in required_attributes:  # this also covers our mandatory ones right?
        cli_utils.error("message 1")
    elif arg not in optional_attributes:
        cli_utils.error("message 2")
```